src/builder/QuarkBuilder.sol

```diff
@@ -164,22 +155,20 @@ contract QuarkBuilder {
         }
 
         // Then, transferIntent `amount` of `assetSymbol` to `recipient`
-        if (payment.isToken) {
-            // wrap around paycall
-        } else {
-            (quarkOperations[actionIndex], actions[actionIndex]) = Actions.transferAsset(
-                Actions.TransferAsset({
-                    chainAccountsList: chainAccountsList,
-                    assetSymbol: transferIntent.assetSymbol,
-                    amount: transferIntent.amount,
-                    chainId: transferIntent.chainId,
-                    sender: transferIntent.sender,
-                    recipient: transferIntent.recipient,
-                    blockTimestamp: transferIntent.blockTimestamp
-                })
-            );
-            actionIndex++;
-        }
+        (quarkOperations[actionIndex], actions[actionIndex]) = Actions.transferAsset(
+            Actions.TransferAsset({
+                chainAccountsList: chainAccountsList,
+                assetSymbol: transferIntent.assetSymbol,
+                amount: transferIntent.amount,
+                chainId: transferIntent.chainId,
+                sender: transferIntent.sender,
+                recipient: transferIntent.recipient,
+                blockTimestamp: transferIntent.blockTimestamp
+            }),
+            payment
+        );
```

Comment:
Nice, it looks cleaner to move the paycall wrapping to the `Actions` helper functions.